src/core/image.js

```diff
@@ -343,6 +351,69 @@ class PDFImage {
     return { data, width, height, interpolate };
   }
 
+  static createMask({
+    imgArray,
+    width,
+    height,
+    imageIsFromDecodeStream,
+    inverseDecode,
+    interpolate,
+  }) {
+    if (
+      typeof PDFJSDev === "undefined" ||
+      PDFJSDev.test("!PRODUCTION || TESTING")
+    ) {
+      assert(
+        imgArray instanceof Uint8ClampedArray,
+        'PDFImage.createMask: Unsupported "imgArray" type.'
+      );
+    }
+
+    const isSingleOpaquePixel =
+      width === 1 &&
+      height === 1 &&
+      inverseDecode === (imgArray.length === 0 || !!(imgArray[0] & 128));
+
+    if (isSingleOpaquePixel) {
+      return { isSingleOpaquePixel };
+    }
+
+    if (FeatureTest.isOffscreenCanvasSupported) {
+      const canvas = new OffscreenCanvas(width, height);
+      const ctx = canvas.getContext("2d");
+      const imgData = ctx.createImageData(width, height);
+      applyMaskImageData({
+        src: imgArray,
+        dest: imgData.data,
+        width,
+        height,
+        inverseDecode,
+      });
+
+      ctx.putImageData(imgData, 0, 0);
+      const bitmap = canvas.transferToImageBitmap();
+
```

Comment:
At this point, I'm guessing that the OffscreenCanvas is no longer needed for anything once the `ImageBitmap` has been created?
If so, would it make sense to try and "dispose" of it *immediately* similar to what we do in other parts of the code-base; see e.g. https://github.com/mozilla/pdf.js/blob/2b673a694114268bad7cc7dd687bf004513e1815/web/pdf_page_view.js#L297-L300

Reply:
I'm not sure it's useful, because we transfered the ownership of the underlying surface (in the ImageBitmap) associated with this canvas and the object itself should be destroyed at the end of the scope.
In following what happens when e.g. `SetWidth` is called:
https://searchfox.org/mozilla-central/rev/c0a97899404931861a4bbec21f55d3cd79b90bb8/dom/canvas/OffscreenCanvas.cpp#76
I don't see anything special which could indicate that's useful.
Anyway, I trust your judgement so if you think it's better, I'll add these lines.

Reply:
I really have no idea if this matters for OffscreenCanvas, I was simply asking based on a pattern we use elsewhere for "regular" canvases (originally added all the way back in PR #4920).

So, do whatever you see fit here :-)

Reply:
@jrmuizel, do you know if setting width & height to 0 is useful for an `OffscreenCanvas` with a lifetime limited to a function scope ?